File: affaj/src/Lib/Core/Snippet.hs
{-# LANGUAGE DeriveAnyClass #-}

module Lib.Core.Snippet
  ( Snippet (..),
  )
where

import Lib.Core.Id (Id)
import Lib.Core.SnippetContent (SnippetContent)
import Lib.Core.SnippetDescription (SnippetDescription)
import Lib.Core.SnippetTitle (SnippetTitle)
import Lib.Core.User (User)

-- | Data type representing row in the @snippets@ table.
data Snippet = Snippet
  { snippetId :: !(Id Snippet),
    title :: !SnippetTitle,
    author :: !(Id User),
    description :: !SnippetDescription,
    content :: !SnippetContent
  }
  deriving stock (Generic, Show, Eq)
  deriving anyclass (FromRow)
  deriving (FromJSON, ToJSON)

File: packages/types/snippet.ts
import { Id } from './id'
import { User } from './user'

export interface Snippet {
   id: Id<Snippet>
   title: string
   author: Id<User>
   description?: string
   content: string
   created_at: string
}

export interface InsertSnippetDTO {
   title: string
   description?: string
   author: Id<User>
   content: string
}
